File: win/MagPrime/Models/AppSettings.cs
namespace MagPrime.Models;

public sealed class AppSettings
{
    public bool LaunchOnStartup { get; set; } = true;
    public int MaxMenuItems { get; set; } = 12;
    public string SortMode { get; set; } = "Recent";
    public string[] ExcludedProcesses { get; set; } = Array.Empty<string>();
    public bool EnableRestore { get; set; } = false;
    public bool ShowToast { get; set; } = true;
}

File: mac/Sources/MagPrimeMac/Models/AppSettings.swift
import Foundation

struct AppSettings: Codable {
    var launchOnStartup: Bool = true
    var maxMenuItems: Int = 12
    var sortMode: String = "Recent" // "Alphabetic" supported
    var excludedApps: [String] = []
    var enableRestore: Bool = false
    var showToast: Bool = true
}
